src/main/java/org/jabref/model/openoffice/util/OOResult.java

```diff
@@ -0,0 +1,128 @@
+package org.jabref.model.openoffice.util;
+
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+/*
+ * error cannot be null
+ * result cannot be null
+ *
+ * Void is not allowed for R, use OOVoidResult instead.
+ *
+ * Out of `isPresent()` and `isError()` exactly one is true.
+ */
+public class OOResult<R, E> {
+    private final Optional<R> result;
+    private final Optional<E> error;
+
+    /**
+     * Exactly one of the arguments should be Optional.empty()
+     *
+     * @param result
+     * @param error
+     */
+    private OOResult(Optional<R> result, Optional<E> error) {
+        this.result = result;
+        this.error = error;
+    }
+
+    /**
+     * @param result Null is not allowed.
+     */
+    public static <R, E> OOResult<R, E> ok(R result) {
+        return new OOResult<>(Optional.of(result), Optional.empty());
+    }
+
+    /**
+     * @param error Null is not allowed.
+     */
+    public static <R, E> OOResult<R, E> error(E error) {
+        return new OOResult<>(Optional.empty(), Optional.of(error));
+    }
+
+    /*
+     * Test state
+     */
+
+    public boolean isPresent() {
+        return result.isPresent();
+    }
+
+    public boolean isEmpty() {
+        return !isPresent();
+    }
+
+    public boolean isError() {
+        return error.isPresent();
+    }
+
+    public boolean isOK() {
+        return !isError();
+    }
+
+    /*
+     * getters
+     */
+
+    public R get() {
+        if (isError()) {
+            throw new RuntimeException("Cannot get from error");
```

Comment:
Never ever throw a `RuntimeException`. Please use `IllegalStateException` (even though this is also not good, but I see that it is required).

However, this is still not consistent to Optionals itself. You should throw a NullPointerException inssted.

Thus:

```java
    return result.orElseThrow(new NullPointerException("Cannot get from error"));
```

Reply:
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Optional.html#get()

Optional.get() throws NoSuchElementException

changed to `throw new NoSuchElementException("Cannot get from error");` to keep in line with Optional